otlp/common.go

```diff
@@ -580,3 +583,55 @@ func shouldTrimTraceId(traceID []byte) bool {
 	}
 	return true
 }
+
+func GetSampleRate(attrs map[string]interface{}) int32 {
+	sampleRateKey := GetSampleRateKey(attrs)
+	if sampleRateKey == "" {
+		return defaultSampleRate
+	}
+
+	sampleRate := defaultSampleRate
+	sampleRateVal := attrs[sampleRateKey]
+	switch v := sampleRateVal.(type) {
```

Comment:
what do you think about adding `float64` and `float32` to this, in case someone sends a value as '1.0`? This might happen if they're thinking about sampling probability instead of sample rate and have to do a division to get the value.

Reply:
I added a check for floats also in the string conversion. I believe I made the change you suggested but take a look to confirm.